Src/library/net/network/socket/network_socket.e

```diff
@@ -206,35 +238,73 @@ feature -- Status setting
 
 	set_timeout (n: INTEGER)
 			-- Set timeout to `n' seconds.
+		obsolete
+			"Use `set_timeout_ns (ns)` [2018-01-15]"
 		require
 			non_negative: n >= 0
 		do
-			if n > 0 then
-				timeout := n
-			else
-				timeout := default_timeout
-			end
+			set_timeout_ns (n.to_natural_64)
 		ensure
 			timeout_set: timeout = n or timeout = default_timeout
 		end
 
+	set_timeout_ns (a_timeout_nanoseconds: NATURAL_64)
+			-- Set timeout with to `a_timeout_nanoseconds` nanoseconds.
+			-- warning: the timeout granularity of the platform may not be nanoseconds, but micro or milliseconds.
+		require
+			positive_timeout: a_timeout_nanoseconds /= 0
```

Comment:
The tag name may need to be updated.

Reply:
on the other hand ... it is not wrong ... "positive" means > 0 
but ok "non_zero_timeout"